index.js

```diff
@@ -369,6 +369,30 @@
         }
       });
 
+  //  reduce :: Foldable f => (a -> b -> a) -> a -> f b -> a
+  var reduce =
+  def('reduce',
+      {b: [Foldable]},
+      [$.Function, a, b, a],
+      function(f, initial, foldable) {
+        if (_type(foldable) === 'Array') {
+          var acc = initial;
+          for (var idx = 0; idx < foldable.length; idx += 1) {
+            acc = f(acc, foldable[idx]);
+          }
+          return acc;
+          } else {
+          return foldable.reduce(f, initial);
+        }
+      });
+
+  //  sum :: Foldable f => f FiniteNumber -> FiniteNumber
+  var sum =
+  def('sum',
+      {},
+      [$.Array($.FiniteNumber), $.FiniteNumber],
+      reduce(function(a, b) { return a + b; }, 0));
+
```

Comment:
Ah, because `meld` is applied in the definition of `mapMaybe`. Let's change that definition from `meld([R.map, justs])` to something along these lines:

```javascript
function(f, xs) { return justs(R.map(f, xs)); }
```